### PR TITLE
USWDS-Site - Modal: Update accessibility guidance

### LINKED ISSUE #2840: USWDS-Site - Feature: Add modal accessibility guidance for trapping focus
### Is your feature request related to a problem? Please describe.

The modal component code does trap the users' focus inside the modal upon activation.

However there is no callout to ensure that it is a key element to accessibility compliance for that component.

### Describe the solution you'd like

On the modal page https://designsystem.digital.gov/components/modal/ under "accessibility" please add the following:

Ensure keyboard focus is trapped inside the modal. When using a keyboard to test the modal, ensure focus does not move to the page behind it.

### Describe alternatives you've considered

The modal accessibility tests will expand on this guidance but it is important to call out keyboard focus for the modal to highlight its importance.

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://designsystem.digital.gov/about/community/#community-conduct).
- [X] I checked the [current issues](https://github.com/uswds/uswds-site/issues) for duplicate feature requests.

### PR DESCRIPTION
**Pull request by mejiaj**
# Summary

Add focus trap guidance. Updated changelog and [content guidance GDOC 🔒 ](https://docs.google.com/document/d/14dUu6qd9STeUfSQj_LO4xVeLywJiU42gS3LxPI3hIZs/edit)

## Related issue

Closes #2840.

## Preview link

[Preview link  →](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-modal-a11y-focus-guidance/components/modal/#accessibility-guidance)

### CODE REVIEW

**Review by amycole501 — APPROVED** [review]

**Review by finekatie — APPROVED** [review]
LGTM!

**Comment by finekatie** [comment]
Hey @mejiaj , does this need a change log update?

**Comment by mejiaj** [comment]
@finekatie ready for another look, I've updated the [changelog](https://federalist-ead78f8d-8948-417c-a957-c21ec5617a57.sites.pages.cloud.gov/preview/uswds/uswds-site/jm-modal-a11y-focus-guidance/components/modal/#:~:text=2024%2D10%2D22,uswds%2Dsite%232874) and guidance in gdoc.

image

**Review by finekatie — APPROVED** [review]
Looks good!

File: _data/changelogs/component-modal.yml [hunk 1]
```diff
@@ -2,6 +2,14 @@ title: Modal
 type: component
 changelogURL:
 items:
+- date: 2024-10-22
```
Comment (thisisdano) [inline _data/changelogs/component-modal.yml:5]:
```suggestion
- date: 2024-11-6
```

**Review by thisisdano — APPROVED** [review]

**Pull request merged**